horovod/common/ops/hip_operations.cc

```diff
@@ -14,17 +14,18 @@
 // limitations under the License.
 // =============================================================================
```

Comment:
This file is now virtually identical to `ops/cuda_operations.cc` except for `cuda` -> `hip` string replacements and the fact that `ScaleBufferImpl` is not implemented for AMD.

I think it's fine to have the two separate files for now.